### PR TITLE
feat(torii): add transaction hash to erc_transfers table

### PR DESCRIPTION
**Pull request by lambda-0x**
**Stack**:
- #2527
- #2526
- #2520 ⬅

⚠️ *Part of a stack created by [spr](https://github.com/ejoffe/spr). Do not merge manually using the UI - doing so may have unexpected results.*

## Summary by CodeRabbit

- **New Features**
	- Introduced a new function to extract transaction hashes from event IDs, enhancing event processing capabilities.
	- Added a `transaction_hash` field to the `erc_transfers` table for improved data capture during token transfers.

- **Bug Fixes**
	- Updated method signatures across various processors to ensure the transaction hash is utilized correctly.

- **Documentation**
	- Enhanced comments in processing methods to clarify expectations for event ID formatting.

- **Style**
	- Updated field names in mappings to follow camelCase formatting and ensure non-nullable types.

- **Chores**
	- Modified database schema to include the new `transaction_hash` column in the `erc_transfers` table.

### CODE REVIEW

File: crates/torii/core/src/engine.rs [hunk 1]
```diff
@@ -610,6 +610,8 @@ impl<P: Provider + Send + Sync + std::fmt::Debug + 'static> Engine<P> {
         let mut unique_contracts = HashSet::new();
         // Contract -> Cursor
         for (event_idx, event) in events.iter().enumerate() {
+            // NOTE: erc* processors expect the event_id to be in this format to get
+            // transaction_hash:
             let event_id =
                 format!("{:#064x}:{:#x}:{:#04x}", block_number, transaction_hash, event_idx);
```
Comment (glihm) [inline crates/torii/core/src/engine.rs:616]: Couldn't we have a function that encapsulates that and add the `NOTE` as doc? :)

File: crates/torii/core/src/processors/erc20_legacy_transfer.rs [hunk 2]
```diff
@@ -40,18 +40,27 @@ where
...
         _block_number: u64,
         block_timestamp: u64,
-        _event_id: &str,
+        event_id: &str,
         event: &Event,
     ) -> Result<(), Error> {
         let token_address = event.from_address;
         let from = event.data[0];
         let to = event.data[1];
+        let transaction_hash = event_id.split(':').nth(1).unwrap();
```
Comment (glihm) [inline crates/torii/core/src/processors/erc20_legacy_transfer.rs:49]: Same here, the `id` construction could be abstracted and would make it easier to modify in the future. 👍

File: crates/torii/core/src/engine.rs (hunk 1 above)
Comment (lambda-0x) [reply crates/torii/core/src/engine.rs:616]: i think we still need the Note there so if someone wants to change the event_id structure they know about this dependency

**Review by glihm — APPROVED** [review]

**Comment by lambda-0x** [comment]
✓ Commit merged in pull request [#2526](https://github.com/dojoengine/dojo/pull/2526)

**Pull request closed without merging**